packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -1785,3 +1785,291 @@ def test_rule_invoked_by_rodsuser_with_remote_via_irule_fails(self):
+
+            with session.make_session_for_existing_admin() as admin_session:
+                # Ensure that the delay queue is empty before attempting to schedule the tiering rule. This is to
+                # prevent false negatives in a later assertion of the emptiness of the delay queue.
+                admin_session.assert_icommand(["iqstat", "-a"], "STDOUT", "No delayed rules pending")
+
+                time.sleep(self.tier0_time_in_seconds)
+                invoke_storage_tiering_rule(admin_session)
+
+            # Wait until the object migrates to the next tier.
+            delay_assert_icommand(self.admin1, ["ils", "-l", self.object_path], "STDOUT", self.tier1)
+
+            # Ensure that nothing is scheduled in the delay queue. The tiering rule should have completed. If
+            # any failures occurred, it is likely that the delayed rule will be retried so we are making sure
+            # that is not happening here.
+            self.admin1.assert_icommand(["iqstat", "-a"], "STDOUT", "No delayed rules pending")
+
+    def do_one_rodsuser_with_permissions_succeeds_test(self, permission):
+        # This is a basic test to show that objects tier out regardless of what permissions for rodsusers exist on it.
+        with storage_tiering_configured():
+            IrodsController().restart(test_mode=True)
+
+            # TODO{#200): Replace with itouch or istream. Have to use put API due to missing PEP support.
+            self.admin1.assert_icommand(["iput", "-R", self.tier0, self.filename, self.object_path])
+
+            # Give permissions exclusively to a rodsuser (removing permissions for original owner).
+            self.admin1.assert_icommand(["ichmod", permission, self.user1.username, self.object_path])
+            self.admin1.assert_icommand(["ichmod", "null", self.admin1.username, self.object_path])
+
+            with session.make_session_for_existing_admin() as admin_session:
+                # Ensure that the delay queue is empty before attempting to schedule the tiering rule. This is to
+                # prevent false negatives in a later assertion of the emptiness of the delay queue.
+                admin_session.assert_icommand(["iqstat", "-a"], "STDOUT", "No delayed rules pending")
+
+                time.sleep(self.tier0_time_in_seconds)
+                invoke_storage_tiering_rule(admin_session)
+
+            # Wait until the object migrates to the next tier.
+            delay_assert_icommand(self.user1, ["ils", "-l", self.object_path], "STDOUT", self.tier1)
+
+            # Ensure that nothing is scheduled in the delay queue. The tiering rule should have completed. If
+            # any failures occurred, it is likely that the delayed rule will be retried so we are making sure
+            # that is not happening here.
+            self.user1.assert_icommand(["iqstat", "-a"], "STDOUT", "No delayed rules pending")
```

Comment:
Should this be executed by the admin?
Will user1 be able to see delay rules queued up by admin_session?

Reply:
Interestingly, rodsusers can indeed see delay rules which they do not own. I'll update this just in case we decide to change that in the future such that a rodsadmin runs this.

Reply:
yes, that feels... like it might need a revisit...   can't think of a reason rodsuser should be able to see more than their own

Reply:
Now using `self.admin1` to run this command. Leaving this open til we've decided what we want to do about rodsusers seeing delay rules from others using `iqstat -a`.

Reply:
I think the following is a good start:
- rodsadmins can see all delay rules
- Non-rodsadmins can only see their delay rules
- iRODS 5
  - GenQuery2 is updated to honor those rules as well 
  - `iqstat` is updated to use GenQuery2
  - Not sure how we handle this for iRODS 4 - can't change how GenQuery1 works
    - Perhaps we deprecate `iquest` and that's all

Other than seeing the delay rules, what else can a non-admin do with that info? I don't think there's much they can do since deleting or modifying delay rules requires admin privileges. Modifying their own rules only puts their stuff at risk.

Reply:
genquery1 can see all kinds of things b/c permissions were/are only on data objects and collections.  not sure we can fix much of that looking backwards.

Reply:
Sounds like a good plan. As that falls outside of the scope of this PR, I think I will go ahead and resolve this...